python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -299,40 +298,64 @@ cdef class Column:
         )
 
     @staticmethod
-    cdef Column from_rmm_buffer(
-        unique_ptr[device_buffer] buff,
+    def from_rmm_buffer(
+        DeviceBuffer buff,
         DataType dtype,
         size_type size,
         list children,
-        Stream stream=None,
     ):
         """
         Create a Column from an RMM DeviceBuffer.
 
         Parameters
         ----------
-        buff : unique_ptr[DeviceBuffer]
-            Pointer to the data rmm.DeviceBuffer.
+        buff : DeviceBuffer
+            The data rmm.DeviceBuffer.
         size : size_type
             The number of rows in the column.
         dtype : DataType
             The type of the data in the buffer.
         children : list
             List of child columns.
-        stream : Stream, default None
-            The stream to use for the operation.
 
         Notes
         -----
         To provide a mask and null count, use `Column.with_mask` after
         this method.
         """
-        stream = _get_stream(stream)
-        cdef DeviceBuffer rmm_buff = DeviceBuffer.c_from_unique_ptr(
-            move(buff), stream
-        )
-        cdef gpumemoryview data = gpumemoryview(rmm_buff)
+        if dtype.id() in {
+            type_id.INT8,
+            type_id.INT16,
+            type_id.INT32,
+            type_id.INT64,
+            type_id.UINT8,
+            type_id.UINT16,
+            type_id.UINT32,
+            type_id.UINT64,
+            type_id.FLOAT32,
+            type_id.FLOAT64,
+            type_id.BOOL8,
+            type_id.TIMESTAMP_SECONDS,
+            type_id.TIMESTAMP_MILLISECONDS,
+            type_id.TIMESTAMP_MICROSECONDS,
+            type_id.TIMESTAMP_NANOSECONDS,
+            type_id.DURATION_SECONDS,
+            type_id.DURATION_MILLISECONDS,
+            type_id.DURATION_MICROSECONDS,
+            type_id.DURATION_NANOSECONDS,
+            type_id.DECIMAL32,
+            type_id.DECIMAL64,
+            type_id.DECIMAL128,
+        } and len(children) != 0:
```

Comment:
Can we express this check in terms of some of the functions in pylibcudf.traits?